### PR TITLE
Respect Safari safe area

### LINKED ISSUE #2936: Respect safe-area-insets in mobile Safari
On the new iPad Pro, the input text box at the bottom of the screen overlaps with the Home indicator.

If we can move it up by `env(safe-area-inset-bottom)` this should put it above the home indicator.

I'm happy to try this myself and submit a PR.

### PR DESCRIPTION
**Pull request by FryDay**
Prevents iOS notch and home bar from getting in the way.

Fixes #2936.

### CODE REVIEW

File: client/css/style.css [hunk 1]
```diff
@@ -59,6 +59,7 @@ body {
 	-webkit-user-select: none;
 	-moz-user-select: none;
 	-ms-user-select: none;
+	padding: env(safe-area-inset-top) env(safe-area-inset-right) 0 env(safe-area-inset-left);
```
[the remark this answers is not in the document]
Comment (FryDay) [reply client/css/style.css]: Oh crap, forgot to put that in the comments of the PR. Does anyone know where existing padding is coming from? If I add the bottom here, in portrait layout it is doubled, but correct in landscape. With the bottom set to 0 it is correct in portrait but tight to the bottom in landscape. I'm sure I'm missing something but spend hours trying to figure out what's causing it.

[the remark this answers is not in the document]
Comment (xPaw) [reply client/css/style.css]: Probably `#viewport`, but that also depends on the theme.

[the remark this answers is not in the document]
Comment (xPaw) [reply client/css/style.css]: Any luck with this?

[the remark this answers is not in the document]
Comment (FryDay) [reply client/css/style.css]: Oops, ended up getting busy at work and forgot about it. I'll see if I can't give it another look this week.

[the remark this answers is not in the document]
Comment (FryDay) [reply client/css/style.css]: Alright, best I can tell, this is some webkit issue. I got a hold of a Mac to debug on. Hope to have this resolved tonight.

**Comment by FryDay** [comment]
Apple devices are "fun". This finally works as it should without breaking anything else.

**Comment by xPaw** [comment]
Not keen on wrapping the entire viewport in an absolute position. I try to avoid these as much as possible.

Why doesn't it work with padding or margin alone?

**Comment by FryDay** [comment]
I'm not sure if it's a bug on Apple's side or if I'm still missing something, but I've found multiple accounts of people having similar issues. https://github.com/ionic-team/ionic/issues/12948#issuecomment-331947154

I agree that it *should* work with padding alone, but the div will refuses to actually take 100% height. If anyone reading this is a CSS wizard and can offer an alternative to my solution, I'm all ears.

**Comment by xPaw** [comment]
That sucks. I don't have an apple device to test this on.

I wonder how this interacts with the notches on Android devices.

**Comment by xPaw** [comment]
Gonna close this as, as the current support for `viewport-fit=cover` is not very good.

**Pull request closed without merging**